internal/provider/assume_service_account_policy_resource.go

```diff
@@ -0,0 +1,316 @@
+	ServiceAccountID       types.String `tfsdk:"service_account_id"`
+	ProviderID             types.String `tfsdk:"provider_id"`
+	MaximumSessionDuration types.Int64  `tfsdk:"maximum_session_duration"`
+	ClaimConditions        types.Set    `tfsdk:"claim_condition"`
+}
+
+type claimConditionModel struct {
+	Claim    types.String `tfsdk:"claim"`
+	Value    types.String `tfsdk:"value"`
+	Operator types.String `tfsdk:"operator"`
+}
+
+func (r *assumeServiceAccountPolicyResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_assume_service_account_policy"
+}
+
+func (r *assumeServiceAccountPolicyResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manages an Assume Service Account Policy in Scalr.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of the Assume Service Account Policy.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the Assume Service Account Policy.",
+				Required:    true,
+			},
+			"service_account_id": schema.StringAttribute{
+				Description: "The ID of the Service Account to which this policy is attached.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"provider_id": schema.StringAttribute{
+				Description: "The ID of the Workload Identity Provider associated with this policy.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"maximum_session_duration": schema.Int64Attribute{
+				Description: "The maximum session duration in seconds for the assumed role.",
+				Optional:    true,
+				Computed:    true,
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"claim_condition": schema.SetNestedBlock{
+				Description: "A set of claim conditions for the policy.",
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"claim": schema.StringAttribute{
+							Description: "The claim to match.",
+							Required:    true,
+						},
+						"value": schema.StringAttribute{
+							Description: "The value to match for the claim.",
+							Required:    true,
+						},
+						"operator": schema.StringAttribute{
+							Description: "The operator to use for matching the claim value. Must be one of: 'eq', 'like', 'startswith', or 'endswith'.",
+							Optional:    true,
+							Computed:    true,
+							Validators: []validator.String{
+								stringvalidator.OneOf("eq", "like", "startswith", "endswith"),
+							},
+						},
+					},
+				},
+				Validators: []validator.Set{
+					setvalidator.SizeBetween(1, 10),
```

Comment:
https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-validators/setvalidator@v0.15.0#SizeBetween
> SizeBetween returns an AttributeValidator which ensures that any configured attribute or function parameter value:
> 
> Is a Set.
> Contains at least min elements and at most max elements.
> **Null (unconfigured) and unknown (known after apply) values are skipped.**

So the validator actually passes when no `claim_condition` block is set.
This should be:

```suggestion
					setvalidator.IsRequired(),
					setvalidator.SizeAtMost(10),
```